### PR TITLE
feat: proposed api enable

### PR DESCRIPTION
**Pull request by dimacodota**

### CODE REVIEW

File: src/provideInlineCompletionItems.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        position,
+        limited: response?.is_locked,
+      },
+    ],
+    command: COMPLETION_IMPORTS,
+    title: "accept completion",
+  };
+}
+
+function calculateRange(
```
Comment (yonip23) [inline src/provideInlineCompletionItems.ts:62]: Now that binary version 4.0.36 is released its worth checking that snippets are working - this version includes the correct suffixes calculation.

File: src/globals/proposedAPI.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import showMessage from "../preRelease/messages";
+
+const EXTENSION_ID = "tabnine.tabnine-vscode";
+
+export default async function enableProposed(): Promise<boolean> {
+  return handleProposed().catch(() => false);
+}
+
+async function handleProposed(): Promise<boolean> {
+  const productFilePath = path.join(vscode.env.appRoot, "product.json");
```
Comment (yonip23) [inline src/globals/proposedAPI.ts:13]: Can be a global const

File: src/globals/proposedAPI.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { promises as fs } from "fs";
+import * as vscode from "vscode";
+import * as path from "path";
+import showMessage from "../preRelease/messages";
+
+const EXTENSION_ID = "tabnine.tabnine-vscode";
+
+export default async function enableProposed(): Promise<boolean> {
+  return handleProposed().catch(() => false);
```
Comment (yonip23) [inline src/globals/proposedAPI.ts:9]: Maybe worth logging the error here?

File: src/provideInlineCompletionItems.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import * as vscode from "vscode";
+import { AutocompleteResult, ResultEntry } from "./binary/requests/requests";
+import { completionIsAllowed } from "./provideCompletionItems";
+import runCompletion from "./runCompletion";
+import { COMPLETION_IMPORTS } from "./selectionHandler";
+
+const INLINE_REQUEST_TIMEOUT = 3000;
```
Comment (yonip23) [inline src/provideInlineCompletionItems.ts:7]: I believe you meant `EMPTY_LINE`

File: src/provideInlineCompletionItems.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  position: vscode.Position
+): Promise<vscode.InlineCompletionList> {
+  try {
+    if (!completionIsAllowed(document, position)) {
+      return new vscode.InlineCompletionList([]);
+    }
+
+    const isEmptyLine = document.lineAt(position.line).text.trim().length === 0;
+
+    const response = await runCompletion(
```
Comment (yonip23) [inline src/provideInlineCompletionItems.ts:20]: I had a code here that sends the request twice on empty line - where did it go

Comment (dimacodota) [reply src/provideInlineCompletionItems.ts:20]: this code was taken from [here](https://github.com/codota/tabnine-vscode/pull/706/files#diff-88c55a307a2aafd4bd44fa02d2ceb1a8fffa4246cde36e7cf4981e2df04b8e75R9) , where is the code u r referring to ?

File: src/test/suite/retry.util.test.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import { expect } from "chai";
+import * as sinon from "sinon";
+import { beforeEach, describe, it } from "mocha";
+import retry from "../../utils/retry";
+
+describe("should retry", () => {
+  beforeEach(() => {
```
[the remark this answers is not in the document]
Comment (yonip23) [reply src/test/suite/retry.util.test.ts:7]: this is a really bad suggestion
@yahave

**Pull request merged**